src/ng/directive/attrs.js

```diff
@@ -341,22 +341,35 @@
 
 var ngAttributeAliasDirectives = {};
 
-
 // boolean attrs are evaluated
 forEach(BOOLEAN_ATTR, function(propName, attrName) {
   // binding to multiple is not supported
   if (propName == "multiple") return;
 
-  var normalized = directiveNormalize('ng-' + attrName);
+  function defaultLinkFn(scope, element, attr) {
+    scope.$watch(attr[normalized], function ngBooleanAttrWatchAction(value) {
+      attr.$set(attrName, !!value);
+    });
+  }
+
+  var normalized = directiveNormalize('ng-' + attrName),
+    linkFn = defaultLinkFn;
```

Comment:
So `linkFn` would have it's own `var`? I'm asking because it differs with other places I saw in the code. Just let me know if that's indeed the case and I'll change it.